django/utils/translation/trans_real.py

```diff
@@ -43,7 +43,9 @@
     re.IGNORECASE
 )
 
-language_code_prefix_re = _lazy_re_compile(r'^/(\w+([@-]\w+)?)(/|$)')
+language_code_prefix_re = _lazy_re_compile(
+    r'^/([A-Za-z]{2,4}(-[A-Za-z]{4})?(-([A-Za-z]{2}|\d{3}))?)(/|$)'
+)
```

Comment:
Do we need to be so restrictive? There are many language tags in the [IANA](http://www.iana.org/assignments/language-subtag-registry/language-subtag-registry) registry that don't match this regex, e.g. `i-mingo`, `de-CH-1996`, `de-1996`, or `kl-tunumiit`.

Reply:
changed to less restrictive version. 
I was thinking about implementing regex based on [bcp47](https://www.rfc-editor.org/rfc/bcp/bcp47.txt), but I have doubts since it will be a very long complex regex